pods/pods.go

```diff
@@ -0,0 +1,237 @@
+// Package pods provides pod status for a namespace using kubectl.
+// It supports both a snapshot as well as a continuous stream.
+package pods
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+)
+
+// A PodStatus represents the status of a single Pod.  Error is only set by
+// WatchPodStatus when there was an error decoding the status.
+type PodStatus struct {
+	Name       string // name of pod
+	Namespace  string
+	Containers []ContainerStatus
+	Phase      string // current phase
+	Error      error  // If there was an error
+}
+
+type ContainerStatus struct {
+	Name    string // name of pod
+	Ready   bool   // true if ready
+	Reason  string // if not empty, the reason it isn't ready
+	Message string // human readable reason
+}
+
+func (s PodStatus) String() string {
+	if s.Error != nil {
+		return "Error: " + s.Error.Error()
+	}
+	return fmt.Sprintf("%s: Phase=%s %v", s.Name, s.Phase, s.Containers)
+}
+
+var testData []byte
+
+// GetPodStatus returns the status of the pods found in the supplied namespace.
+func GetPodStatus(namespace string) ([]PodStatus, error) {
+	var data []byte
+	if testData == nil {
+		var stdout bytes.Buffer
+		args := []string{"get", "pods", "-o", "json"}
```

Comment:
should be able to avoid execing here using a kube client:

pods, err := kClient.CoreV1().Pods(namespace).List(context.Background(), metav1.ListOptions{})